### ISSUE TITLE
pass url into the razor view

### ISSUE THREAD

**Issue by marvel420** [Antaris/RazorEngine#362]
Hi -

I am trying to use this tool to generate body of an email which is a razor view.  The view expects a url to be passed in which looks like: http://localhost:1982/#/login/ 
I store that url in my model.

in my razor view, I reference it in this manner:
<a href="@Model.LoginURL"><u>Home Page</u></a>

for some reason the RunCompile funtion dies stating the following error.
	 - error: (31, 13) The name 'BeginWriteAttribute' does not exist in the current context
	 - error: (4, 1) The name 'WriteAttributeValue' does not exist in the current context
	 - error: (37, 13) The name 'EndWriteAttribute' does not exist in the current context

The RunCompile function works fine if I hard code the same url inside the razor view but that is not the requirement. <a href="http://localhost:1982/#/login/"><u>Home Page</u></a>

Any insight in this is much appreciated.

Thanks

**Comment by layinka** [comment]
I am having this very same issue. I saw on stackoverflow to add a basetemplatetype to config, which i have done to no avail

**Comment by bragma** [comment]
Same problem here, I'm stuck trying to reference a model property inside an img src attribute. Any suggestion from the developer would be appreciated!

**Comment by bragma** [comment]
On a quick glance a the code, it seems that the razor compiler generates a source code for the template containing a call to the method 'BeginWriteAttribute'. Since the class containing the template inherits from TemplateBase, that class should implement it, but there is no declaration of the 'BeginWriteAttribute' method in it. I am puzzled...

**Comment by aviatrix** [comment]
Some code samples would be quite helpful.

**Comment by bragma** [comment]
At the moment, the only workaround I've found is to create a string with the HTML tag containing the attribute, and then place the attribute in the template using @Raw(xx). As an example:

@{ 
	var pictureTag = "<img src=\"" + Model.PictureUri + "\" />";
}

<div> Here is the image: @Raw(pictureTag)</div>

**Comment by bragma** [comment]
@aviatrix do you mean sample code showing the problem? I think I can setup a sample project and put it on github if it can help. I can also try to help solving it, but I am not sure I have fully understood the problem. 'BeginWriteAttribute' is a valid method of the class RazorPage, but I don't know who in the RazorEngine environment should provide it. Should it be the TemplateBase class?

**Comment by aviatrix** [comment]
@bragma yes, that was what i meant. please ping me when you get a sample that demonstrates the project

**Comment by matthid** [comment]
Does this work with https://www.nuget.org/packages/RazorEngine/4.4.1-rc1?
